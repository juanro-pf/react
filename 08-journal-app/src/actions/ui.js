import { types } from "../types/types";

export const setError= (err) => ({
  type: types.uiSetError,
  payload: err
});

export const UnsetError= () => ({
  type: types.uiUnsetError
});

export const startLoading= () => ({
  type: types.uiStartLoading
});

export const finishLoading= () => ({
  type: types.uiFinishLoading
});