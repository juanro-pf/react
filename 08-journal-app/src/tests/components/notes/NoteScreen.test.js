import '@testing-library/jest-dom';
import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { activeNote } from '../../../actions/notes';
import { NoteScreen } from '../../../components/notes/NoteScreen';

jest.mock('../../../actions/notes', () => ({
  activeNote: jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState= {
  auth: {
    uid: 'asdasd',
    name: 'Jro'
  },
  ui: {
    loading: false,
    msgError: null
  },
  notes: {
    notes: [],
    active: {
      id: 1234,
      title: 'hola',
      body: 'mudo',
      date: 0
    }
  }
};

let store= mockStore(initState);
store.dispatch= jest.fn();

const wrapper= mount(
  <Provider store= {store}>
    <NoteScreen />
  </Provider>
);

describe('Testing NoteScreen', () => {
  
  test('should match snapshot', () => {
    
    expect(wrapper).toMatchSnapshot();

  });

  test('should dispatch activeNote', () => {
    
    wrapper.find('input[name="title"]').simulate('change', {
      target: {
        name: 'title',
        value: 'Woliii'
      }
    });

    expect(activeNote).toHaveBeenCalledWith(
      1234,
      {
        body: 'mudo',
        title: 'Woliii',
        id: 1234,
        date: 0
      }
    );

  });
  
});