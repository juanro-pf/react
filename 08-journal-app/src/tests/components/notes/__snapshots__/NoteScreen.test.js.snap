// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NoteScreen should match snapshot 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      22 de agosto de 2020
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="hola"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today"
      value="mudo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
